.github/workflows/bazelBuildAndTestLlvm.yml

```diff
@@ -0,0 +1,82 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+name: Bazel Build and Test (llvm-project)
+
+# Only run when llvm-project hash changes (deps.bzl)
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'deps.bzl'
```

Comment:
this is so the workflow is triggered only when change is made to `deps.bzl`, as otherwise we shouldn't have to re-run a workflow (3p test status shouldn't change)